Comments/Services/SubmitService.cs

```diff
@@ -64,6 +66,57 @@ public SubmitService(ConsultationsContext context, IUserService userService, ICo
 			return (rowsUpdated: _context.SaveChanges(), validate: null);
 		}
 
+		public (int rowsUpdated, Validate validate, ConsultationsContext context) SubmitToLead(SubmissionToLead submission)
+		{
+			if (!_currentUser.IsAuthenticatedByOrganisationCookie)
+				return (rowsUpdated: 0, validate: new Validate(valid: false, unauthenticated: true, message: $"User not Authenticated"), null);
+
+			var anySourceURI = submission.SourceURIs.FirstOrDefault();
+			if (anySourceURI == null)
+				return (rowsUpdated: 0, validate: new Validate(valid: false, unauthorised: false, message: "Could not find SourceURI"), null);
+
+			var consultationState = _consultationService.GetConsultationState(anySourceURI, PreviewState.NonPreview);
+
+			if (!consultationState.ConsultationIsOpen)
+				return (rowsUpdated: 0, validate: new Validate(valid: false, unauthorised: false, message: "Consultation is not open for submissions"), null);
+
+			var hasSubmitted = _consultationService.GetSubmittedDate(anySourceURI);
+			if (hasSubmitted != null)
+				return (rowsUpdated: 0, validate: new Validate(valid: false, unauthorised: false, message: "User has already submitted."), null);
+
+			var consultationId = ConsultationsUri.ParseConsultationsUri(anySourceURI).ConsultationId;
+			var organisationUserId = _currentUser.ValidatedSessions.FirstOrDefault(session => session.ConsultationId.Equals(consultationId))?.OrganisationUserId;
+
+			var submissionToSave = _context.InsertSubmission(_currentUser.UserId, null, null, null, null, null);
+
+			var submittedStatus = _context.GetStatus(StatusName.Submitted);
+
+			_context.UpdateEmailAddressForOrganisationUser(submission.EmailAddress, (int)organisationUserId);
+			if (submission.Comments.Count > 0)
+			{
+				UpdateCommentsModel(submission.Comments, submissionToSave, submittedStatus);
+				var commentIds = submission.Comments.Select(c => c.CommentId).ToList();
+				_context.DuplicateComment(commentIds);
+			}
+			if (submission.Answers.Count > 0)
+			{
+				UpdateAnswersModel(submission.Answers, submissionToSave, submittedStatus);
+				var answerIds = submission.Answers.Select(a => a.AnswerId).ToList();
+				_context.DuplicateAnswer(answerIds);
+			}
+
+			//now for analytics calculate the number of seconds between the user's first comment or answer and the submission date
+			var earliestDate = submissionToSave.SubmissionComment.Any() ? submissionToSave.SubmissionComment.Min(sc => sc.Comment.CreatedDate) : DateTime.MaxValue;
```

Comment:
is there any chance this block of code could be shared between the submissions please?

It's just quite a chunk of code that looks identical to me.

so it'd be something like:

Submission.DurationBetweenFirstCommentOrAnswerSavedAndSubmissionInSeconds = GetDuration(submissionToSave);

then all that code could be moved into a GetDuration function.